src/classes/TDTM_TriggerHandler.cls

```diff
@@ -153,63 +145,121 @@ public class TDTM_TriggerHandler {
+                } else {
+                    UTIL_Debug.debug(LoggingLevel.WARN, '****The class does not implement the required interface.');
+                }
+            } else {
+                UTIL_Debug.debug(LoggingLevel.WARN, '****ClassType was null.');
+            }
         } else {
-        	UTIL_Debug.debug(LoggingLevel.WARN, '****ClassToCall was null');
+            UTIL_Debug.debug(LoggingLevel.WARN, '****ClassToCall was null');
         }
         return null;
     }
-    
-    private static void runAsync(TDTM_Runnable classToRun, String classToRunName, List<Sobject> newList, 
-    List<Sobject> oldList, TDTM_Runnable.Action thisAction, Schema.DescribeSobjectResult describeObj) {
-       set<Id> setNewId;
-       if(newlist != null) {
-           Map<Id,Sobject> nm = new Map<Id,Sobject>(newlist);
-           setNewId = new set<Id>(nm.keySet());
+
+    /*******************************************************************************************************
+     * @description Given a specific TriggerHandler record linked to the current trigger event, execute the
+     * specified class based on the type of class instance it represents.
+     * @param classToRun Instance of class that implements the TDTM_Runnable interface
+     * @param classToRunName The name of the class to be executed
+     * @param newList Collection of Trigger.New
+     * @param oldList Collection of Trigger.Old
+     * @param thisAction Trigger Event (Insert, Update, Delete, ...)
+     * @param describeObj SObjectDescribe of the target object
+     */
+    private static void runAsync(TDTM_Runnable classToRun, String classToRunName, List<SObject> newList,
+            List<SObject> oldList, TDTM_Runnable.Action thisAction, Schema.DescribeSObjectResult describeObj) {
+
+       Set<Id> setNewId;
+       if (newList != null) {
+           Map<Id,SObject> nm = new Map<Id,SObject>(newList);
+           setNewId = new Set<Id>(nm.keySet());
        }
        
-       set<Id> setOldId;
-       if(oldlist != null) {
-           Map<Id,Sobject> om = new Map<Id,SObject>(oldlist);
-           setOldId = new set<Id>(om.keySet());
+       Set<Id> setOldId;
+       if (oldList != null) {
+           Map<Id,SObject> om = new Map<Id,SObject>(oldList);
+           setOldId = new Set<Id>(om.keySet());
        }
        
        classToRun.runFutureNonStatic(setNewId, setOldId, thisAction.name(), describeObj.getName(), classToRunName);
     }
+
+    /**
+     * @description Combine the DmlWrapper instance returned from an individual TriggerHandler run() call into
+     * the DmlWrapper instance for the entire TriggerHandler.run() method.
+     * @param dmlWrapper The shared instance
+     * @param dmlWrapperLocal The single instance returned from a TriggerHandler call.
+     */
+    private static void updateDmlWrapperWithTriggerResponse(TDTM_Runnable.DmlWrapper dmlWrapper, TDTM_Runnable.DmlWrapper dmlWrapperLocal) {
```

Comment:
Nice pulling this out into it's own method, small incremental clean up like this goes a long way!